### DISCUSSION TITLE
Is there a way to control zIndex across ShapeSources

### DISCUSSION THREAD

**Discussion by bradstiff** [rnmapbox/maps#2639] [Q&A]
I have a map with multiple ShapeSources. I am able to control the zIndex of the ShapeSources via the order in which the components are specified, e.g., in the example below, `<ShapeSourceB />` is rendered on top of `<ShapeSourceA />`

```js
<MapView>
  <ShapeSourceA />
  <ShapeSourceB />
</MapView>
```
The problem is the user can toggle between two map layers. In the example below, `mapLayer` defaults to 'mapLayer1.' When the user toggles `<MapLayer2 />`, it gets added to the map and rendered on top of `<ShapeSourceB />`.

```js
<MapView>
  <ShapeSourceA />
  {mapLayer === 'mapLayer1' && <MapLayer1 />}
  {mapLayer === 'mapLayer2' && <MapLayer2 />}
  <ShapeSourceB />
</MapView>
```

As a workaround, I'm doing the following, but I wonder if there is a better way?

```js
<MapView>
  <ShapeSourceA />
  {mapLayer === 'mapLayer1' && <>
    <MapLayer1 />
    <ShapeSourceB />
  </>}
  {mapLayer === 'mapLayer2' && <>
    <MapLayer2 />
    <ShapeSourceB />
  </>}
</MapView>
```

**Comment by mfazekas** [comment]
@bradstiff sources doesn't have orders. Layers do. There is some aboveLayerID and belowLayerID to control [order of layers](https://github.com/rnmapbox/maps/issues/1477).

**Comment by bradstiff** [comment]
I ended up doing the following:

```javascript
<MapView>
  <ShapeSourceA />
  <MapLayer1 visible={mapLayer === 'mapLayer1'} />
  <MapLayer2 visible={mapLayer === 'mapLayer2'} />
  <ShapeSourceB />
</MapView>
```

The `MapLayer` components render an empty shape when `visible = false`.

Does this seem safe? E.g., is it reliable to leverage the order in which the `Shapes` are initially rendered to control zOrder?
